### ISSUE TITLE
Training is not reproducible between Intel and ARM (M2)

### ISSUE THREAD

**Issue by lusis-ai** [google/yggdrasil-decision-forests#92]
When doing the same simple RF classifier training on an Intel Xeon and on Mac M2, metrics are different (and of course trees are also different).

```python
import ydf  # Yggdrasil Decision Forests
import pandas as pd  # We use Pandas to load small datasets

# Download a classification dataset and load it as a Pandas DataFrame.
ds_path = "https://raw.githubusercontent.com/google/yggdrasil-decision-forests/main/yggdrasil_decision_forests/test_data/dataset"
train_ds = pd.read_csv(f"{ds_path}/adult_train.csv")
test_ds = pd.read_csv(f"{ds_path}/adult_test.csv")


model = ydf.RandomForestLearner(label="income",
                                task=ydf.Task.CLASSIFICATION).train(train_ds)


evaluation = model.evaluate(test_ds)

print(evaluation)
```

Accuracy are :

- Intel: 0.866005
- M2 Max : 0.866107

The other metrics and the threes are also different.

The same problem happened on Scikit-Learn and XGBoost when setting some hyper parameters using random number generation. It acts like if the random sequence is not the same between Intel and ARM.

But it does not happened with LightGBM that is the only one enabling perfect reproducibility no matter the CPU.

**Comment by kjyv** [comment]
I've seen similar behaviour also with scikit-learn's implementation. It's likely pretty normal to have different accuracies, as there are differences in the floating point implementations of the CPUs. Your accuracy differences are relatively small compared to what we're seeing (~1%). The PRNG should produce the same numbers, but the thresholds, impurities and averaging for the predictions will be slightly different.

**Comment by rstz** [comment]
You're correct that YDF models can vary across platforms due to differences in the underlying randomness primitives. YDF uses C++'s <random> and occasionally Abseil's random library, neither of which guarantees identical output across platforms, even with the same seed. Abseil details this here: https://abseil.io/docs/cpp/guides/random#seed-stability. This can lead to slight model differences when training on different systems.

While strict cross-platform reproducibility hasn't been prioritized in the past, we're gathering use cases to understand needs. If you run into a use case for it, please comment on this issue.
